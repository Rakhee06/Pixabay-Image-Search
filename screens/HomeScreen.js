import React, { Component } from 'react';
import { View,
    StyleSheet
} from 'react-native';
import { Header,
    Input,
    Button
} from 'react-native-elements';
import { Icon } from 'react-native-elements';

import { MonoText } from '../components/StyledText';

export default class HomeScreen extends Component {
    static navigationOptions = {
        title: 'Home',
    };

    state = {
        search: ''
    };

    // Function to navigate to the next screen with search text
    onSubmitHandler= () => {
        const response = this.props.navigation.navigate('List', {
                event: this.state.search
            })
    };
    render() {
        return (
            <View style={styles.container}>
                <Header
                    containerStyle = {{ backgroundColor: '#478692' }}
                    leftComponent={{ color: '#fff' }}
                    centerComponent={<MonoText style={{ color: '#fff' }}>
                        PIXABAY IMAGE SEARCH</MonoText>}>
                </Header>
                <View style={styles.textContainerStyle}>
                    <Input
                        placeholder='Type here..'
                        containerStyle={styles.textInputStyle}
                        onChangeText={search => this.setState({search})}
                        value={this.state.search}
                        onSubmitEditing={this.onSubmitHandler}
                    />
                    <Button
                        buttonStyle={styles.buttonStyle}
                        icon={
                            <Icon
                                name='search1'
                                type='antdesign'
                                color='#fff'
                            />
                        }
                        onPress={this.onSubmitHandler}
                    />
                </View>
            </View>
    );
  };
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'lightgray',
    },
    textContainerStyle: {
        flexDirection: 'row',
        marginTop: 75,
        marginLeft: 10,
        marginRight: 10,
    },
    textInputStyle: {
        flex: 1,
    },
    buttonStyle: {
        marginBottom: 8,
        backgroundColor: '#478692',
    }
});