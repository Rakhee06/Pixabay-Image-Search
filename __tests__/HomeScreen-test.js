import 'react-native';
import React from 'react';
import HomeScreen from '../screens/HomeScreen';
import renderer from 'react-test-renderer';

describe("Home Screen", () => {
    it('renders the home screen correctly', async () => {
        const tree = await renderer.create(<HomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});