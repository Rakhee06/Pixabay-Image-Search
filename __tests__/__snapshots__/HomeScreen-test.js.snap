// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders the home screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightgray",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#478692",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <Text
        color="#fff"
        style={Object {}}
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontFamily": "space-mono",
            },
          ]
        }
      >
        PIXABAY IMAGE SEARCH
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 75,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Type here.."
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#478692",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 8,
              "padding": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
